### ISSUE TITLE
Transpose axes not matching error when rotating by 90 or -90 degrees

### ISSUE THREAD

**Issue by OneWorld-github** [Zulko/moviepy#1278]
```
import moviepy.editor as mped
import numpy as np
colour_clip = mped.ColorClip(size=[300, 200], color=np.array([250, 90, 0]).astype(np.uint8), duration=3).set_position((0, 0))
rot_trans_col_clip = colour_clip.add_mask().rotate(90, unit='deg', expand=True, resample='bilinear')
stacked_clips = mped.CompositeVideoClip([rot_trans_col_clip]).set_duration(3)
stacked_clips.write_videofile('90_degrees_in_moviepy.mp4', fps=5)
```

#### Expected Behavior

a 90 degree rotated asset

#### Actual Behavior

```
Traceback (most recent call last):
  File "c:/..../pillow-rotation.py", line 16, in <module>
    rot_trans_col_clip = colour_clip.add_mask().rotate(90, unit='deg', expand=True, resample='bilinear')
  File "C:\....\Python38\lib\site-packages\moviepy\video\fx\rotate.py", line 71, in rotate
    return clip.fl(fl, apply_to=["mask"])
  File "C:\....\Python38\lib\site-packages\moviepy\video\VideoClip.py", line 943, in fl
    newclip = VideoClip.fl(self, fl, apply_to=apply_to,
  File "C:\....\Python38\lib\site-packages\moviepy\Clip.py", line 151, in fl
    new_a = a.fl(fun, keep_duration=keep_duration)
  File "C:\....\Python38\lib\site-packages\moviepy\video\VideoClip.py", line 943, in fl
    newclip = VideoClip.fl(self, fl, apply_to=apply_to,
  File "C:\....\Python38\lib\site-packages\moviepy\Clip.py", line 138, in fl
    newclip = self.set_make_frame(lambda t: fun(self.get_frame, t))
  File "<decorator-gen-57>", line 2, in set_make_frame
  File "C:\....\Python38\lib\site-packages\moviepy\decorators.py", line 14, in outplace
    f(newclip, *a, **k)
  File "C:\....\Python38\lib\site-packages\moviepy\video\VideoClip.py", line 666, in set_make_frame
    self.size = self.get_frame(0).shape[:2][::-1]
  File "<decorator-gen-10>", line 2, in get_frame
  File "C:\....\Python38\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\....\Python38\lib\site-packages\moviepy\Clip.py", line 95, in get_frame
    return self.make_frame(t)
  File "C:\....\Python38\lib\site-packages\moviepy\Clip.py", line 138, in <lambda>
    newclip = self.set_make_frame(lambda t: fun(self.get_frame, t))
  File "C:\....\Python38\lib\site-packages\moviepy\video\fx\rotate.py", line 59, in fl
    return np.transpose(im, axes=transpo)[::-1]
  File "<__array_function__ internals>", line 5, in transpose
  File "C:\....\Python38\lib\site-packages\numpy\core\fromnumeric.py", line 653, in transpose
    return _wrapfunc(a, 'transpose', axes)
  File "C:\....\Python38\lib\site-packages\numpy\core\fromnumeric.py", line 58, in _wrapfunc
    return bound(*args, **kwds)
ValueError: axes don't match array
```

#### Steps to Reproduce the Problem
copy the code above and paste into an IDE, and run the code

#### Specifications

  - Python Version: 3.8.4
  - Moviepy Version: 1.0.0
  - Platform Name: Windows
  - Platform Version: 10

**Comment by tburrows13** [comment]
Update Moviepy to 1.0.3 and if that doesn't fix it, try 2.0.0.dev1 :)

**Comment by OneWorld-github** [comment]
tried both, neither of these worked.  Got a fix suggestion / workaround Tom ?

**Comment by tburrows13** [comment]
Oh that's interesting, I thought that I'd fixed that. I'll have to investigate then, but I don't know when I'll be able to.

Why are you creating and then rotating the colorclip though? Why not just create it with the reversed sizes?

**Comment by tburrows13** [comment]
Alternatively edit the code to remove the special case for multiples of 90.

**Comment by OneWorld-github** [comment]
> Oh that's interesting, I thought that I'd fixed that. I'll have to investigate then, but I don't know when I'll be able to.
> 
> Why are you creating and then rotating the colorclip though? Why not just create it with the reversed sizes?

colorclip was just an example, it's the same error with an image aswell.

How would you go about doing a transpose of the image to 90 degrees, and update the values that have changed in the clip object, for continued process in moviepy ?

**Comment by OneWorld-github** [comment]
> Alternatively edit the code to remove the special case for multiples of 90.

what if moviepy is reinstalled ?  Then edits will be lost

**Comment by tburrows13** [comment]
Sure, translating an image clip would be harder, but luckily the code is already there in rotate.py :) if you copy that then it will work even if moviepy is reinstalled, whereas, yes, if you edit the source then that is only a temporary fix. Of course, both of these are only workarounds until it is properly fixed.

**Comment by OneWorld-github** [comment]
> Sure, translating an image clip would be harder, but luckily the code is already there in rotate.py :) if you copy that then it will work even if moviepy is reinstalled, whereas, yes, if you edit the source then that is only a temporary fix. Of course, both of these are only workarounds until it is properly fixed.

Are you suggesting something like this...
If angle = 90:
   transposed_image = numpy.rot90(extracted_image_from_clip_object, 1)
   transposed_clip = mped.ImageClip(transposed_image, duration=vid_duration).set_position((x, y))
else:
   clip.rotate(90)

if so how do i extract the image from the clip object ?

**Comment by tburrows13** [comment]
Sorry for the delay, but yes I think that would work. The image as a numpy array should be accessible from an ImageClip with `clip.img`.

The even simpler option imo would be to take the code from [rotate.py](https://github.com/Zulko/moviepy/blob/master/moviepy/video/fx/rotate.py) and just redefine a modified version yourself. Note that it has special cases for when the angle is a multiple of 90, and these are the bits that don't work. It has a fallback method for when the angle isn't a multiple of 90 (which does work for you as you noted that it worked when the angle was 90.0001), so just remove lines 75 to 82.

**Comment by tburrows13** [comment]
Fixed in #1335
